juniper/src/integrations/json.rs

```diff
@@ -0,0 +1,106 @@
+use serde_json::Value as JsonValue;
+
+use parser::{ParseError, ScalarToken, Token};
+use value::ParseScalarResult;
+use Value;
+
+graphql_scalar!(JsonValue as "JsonString" where Scalar = <S> {
+    description: "JSON serialized as a string"
+
+    resolve(&self) -> Value {
+        Value::scalar(self.to_string())
+    }
+
+    from_input_value(v: &InputValue) -> Option<JsonValue> {
+        v.as_scalar_value::<String>()
+         .and_then(|s| serde_json::from_str(s).ok())
+    }
+
+    from_str<'a>(value: ScalarToken<'a>) -> ParseScalarResult<'a, S> {
```

Comment:
My custom scalar implementation that worked with JSON objects looked like this:

```
from_str<'a>(value: ScalarToken<'a>) -> ParseScalarResult<'a, S> {
    <String as ParseScalarValue<S>>::from_str(value)
}
```

Otherwise, it looks pretty much exactly the same. I don't understand the intricacies of `graphql_scalar!` so I am not sure that is the correct way to do it...

Reply:
Just tested this change and it is working for me.